### PR TITLE
Add absolutePath to swagger-ui-dist exports

### PR DESCRIPTION
**Pull request by dylancwood**
In v2.x.x of swagger-ui, there was a `index.js` in the package root. This file exported a `dist` property which was the absolute path to the `dist/` directory. This was useful for nodejs webservers that would serve swagger-ui as a static site. (https://github.com/swagger-api/swagger-ui/blob/v2.2.8/index.js#L8)

In v3.x.x of swagger-ui and swagger-ui-dist, the `index.js` file in the package root does not export such a property. 

This PR adds an `absolutePath` property to the swagger-ui-dist package index.js file. I feel that `absolutePath` is a more descriptive name for the property, especially now that the _dist_ package is separate from the primary core package.

### CODE REVIEW

**Comment by shockey** [comment]
@dylancwood I like the idea, but what happens when you try to use this in a browser?

**Comment by dylancwood** [comment]
I imagine that it would break, since `path` is not likely going to work in a browser context. I can make it more browser compatible.

**Comment by dylancwood** [comment]
@shockey: on second inspection, `index.js` won't run in a browser anyway since browsers do not support module loading via `require` AFAIK. This means that one would need to preprocess `index.js` with something like webpack. 

We could implement the following, which would throw an understandable error if `getAbsolutePath()` were called from within a browser context.
```
module.exports.getAbsolutePath = () => {
  // detect whether we are running in a browser or nodejs
  if (typeof module !== 'undefined' && module.exports) {
    return require('path').resolve(__dirname);
  }
  throw new Error('getAbsolutePath only available when running in NodeJS');
};
```

**Comment by shockey** [comment]
@dylancwood Yep, I'm thinking of someone trying to use the `swagger-ui-dist` module with webpack/browserify/whatever. Your solution sounds good.

Not sure why one would want to use `swagger-ui-dist` with a bundler (just use the regular module...), but I'd prefer to not have the whole thing explode if someone does, without having to assume that any bundler will shim out `path` correctly.

**Comment by dylancwood** [comment]
@shockey:

I've incorporated the changes discussed, with a few tweaks:
1. Performing `require` inside a function that is only executed at run time seems risky, so I encapsulated it in a function, but execute that function right away to ensure that any issues are exposed on initial startup.
2. For testability, I had to move the `absolutePath` logic into a separate file (swagger-ui-dist-package/index.js cannot be executed in its current form: `deploy.sh` would need to be run first).

File: swagger-ui-dist-package/index.js [hunk 1]
```diff
@@ -1,2 +1,4 @@
-module.exports.SwaggerUIBundle = require('./swagger-ui-bundle.js')
-module.exports.SwaggerUIStandalonePreset = require('./swagger-ui-standalone-preset.js')
+module.exports.SwaggerUIBundle = require("./swagger-ui-bundle.js")
```
Comment (dylancwood) [inline swagger-ui-dist-package/index.js]: Use double quotes to be consistent with the rest of the lib

File: swagger-ui-dist-package/absolute-path.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * getAbsoluteFSPath
+ * @return {string} When run in NodeJS env, returns the absolute path to the current directory
+ *                  When run outside of NodeJS, will return an error message
+ */
+const getAbsoluteFSPath = () => {
+  // detect whether we are running in a browser or nodejs
+  if (typeof module !== "undefined" && module.exports) {
+    return require("path").resolve(__dirname)
+  }
+  return "Error! absolutePath only available when running in NodeJS"
```
Comment (shockey) [inline swagger-ui-dist-package/absolute-path.js]: Any particular reason for this? 

If not, I'd prefer to throw here or use `console.warn`.

Comment (dylancwood) [inline swagger-ui-dist-package/absolute-path.js]: 👍

Comment (dylancwood) [inline swagger-ui-dist-package/absolute-path.js]: @shockey: addressed. LMK what you think.

File: swagger-ui-dist-package/README.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ Use `swagger-ui` instead, if you'd like to have npm install dependencies for you
 To get an absolute path to this directory for static file serving, use the exported `getAbsoluteFSPath` method:
 
 ```javascript
-const swaggerUiAssetPath = require("swagger-ui-dist").getAbsoluteFSPath
+const swaggerUiAssetPath = require("swagger-ui-dist").getAbsoluteFSPath()
```
Comment (dylancwood) [inline swagger-ui-dist-package/README.md]: Thanks!

**Comment by shockey** [comment]
Thanks for the feature @dylancwood, and thanks for being responsive through the review process. We appreciate your effort 💪 

Merging!

**Pull request merged**